### PR TITLE
Upgrade Winstone from 6.7 to 6.10

### PR DESCRIPTION
**Pull request by jglick**
Easily forgotten due to https://github.com/jenkinsci/jenkins/blob/af5a649a470f7fecff38de754df8ee10c9a17c62/.github/dependabot.yml#L22-L25

* https://github.com/jenkinsci/winstone/pull/310
* https://github.com/jenkinsci/winstone/pull/312
* https://github.com/jenkinsci/winstone/pull/313

### Testing done

`jetty:run` sanity check

### Proposed changelog entries

- Upgraded bundled Winstone. Notably, the `--excludeProtocols` option was added.

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7632"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
